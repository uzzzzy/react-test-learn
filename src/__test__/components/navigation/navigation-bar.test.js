import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import NavigationBar from '../../../components/navigation/navigation-bar'

import { render, screen, cleanup } from '@testing-library/react'

import '@testing-library/jest-dom'

afterEach(cleanup)

it('render navigation bar', () => {
    const div = document.createElement('div')
    ReactDOM.render(<NavigationBar></NavigationBar>, div)
})

it('check title', () => {
    render(<NavigationBar></NavigationBar>)
    expect(screen.getByTestId('nav-bar-title')).toHaveTextContent('Home')
})

it('matches snapshot', () => {
    const tree = renderer.create(<NavigationBar />).toJSON()

    expect(tree).toMatchSnapshot()
})
