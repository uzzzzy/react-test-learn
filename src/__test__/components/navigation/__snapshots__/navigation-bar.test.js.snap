// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="fixed top-0 left-0 bg-white w-full p-3"
  data-testid="nav-bar-background"
>
  <div
    className="flex container mx-auto justify-between"
    data-testid="nav-bar-container"
  >
    <div
      data-testid="nav-bar-title"
    >
      Home
    </div>
  </div>
</div>
`;
